src/types.rs

```diff
@@ -117,6 +119,27 @@ impl RowEntry {
     }
 }
 
+impl Eq for RowEntry {}
+
+impl PartialEq<Self> for RowEntry {
+    fn eq(&self, other: &Self) -> bool {
+        self.key == other.key && self.seq == other.seq
+    }
+}
```

Comment:
i am afraid this might not exactly `PartialEq`'s semantic, i suppose that we should put every field of RowEntry to compare it's euqalness.